pkg/storage/replica.go

```diff
@@ -1869,6 +1869,14 @@ func (r *Replica) tryAddWriteCmd(
 			// Set endCmds to nil because they have already been invoked
 			// in processRaftCommand.
 			endCmds = nil
+			if len(propResult.Intents) > 0 {
```

Comment:
A problem is that this isn't always the path that a finished request follows. Original proposals which succeed but are re-proposed and cancelled commands will both leave intents to the GC. Maybe that's OK, although knowing our luck, there'll be some degenerate use case that creates massive amounts of intent garbage. After some thought I'm not coming up with any good solutions. However, let's add a comment to the effect that these kinds of requests will generate GC-ready intents.